Compilation issue on GPU

When compiling latest dev on Juwels
```
cmake -S . -B build -DHiPACE_COMPUTE=CUDA -DAMREX_GPUS_PER_SOCKET=2 -DAMREX_GPUS_PER_NODE=4 -DCMAKE_BUILD_TYPE=Release -DHiPACE_PRECISION=DOUBLE
cmake --build build -j 16
```
I get the following error
```
[ 88%] Building CUDA object CMakeFiles/HiPACE.dir/src/particles/plasma/PlasmaParticleContainer.cpp.o
/p/home/jusers/thevenet1/juwels/src/hipace/src/particles/deposition/PlasmaDepositCurrent.cpp(221): error: An extended __device__ lambda cannot first-capture variable in constexpr-if context

/p/home/jusers/thevenet1/juwels/src/hipace/src/particles/deposition/PlasmaDepositCurrent.cpp(221): error: An extended __device__ lambda cannot first-capture variable in constexpr-if context

/p/home/jusers/thevenet1/juwels/src/hipace/src/particles/deposition/PlasmaDepositCurrent.cpp(221): error: An extended __device__ lambda cannot first-capture variable in constexpr-if context

/p/home/jusers/thevenet1/juwels/src/hipace/src/particles/deposition/PlasmaDepositCurrent.cpp(221): error: An extended __device__ lambda cannot first-capture variable in constexpr-if context

4 errors detected in the compilation of "/p/home/jusers/thevenet1/juwels/src/hipace/src/particles/deposition/PlasmaDepositCurrent.cpp".
gmake[2]: *** [CMakeFiles/HiPACE.dir/build.make:286: CMakeFiles/HiPACE.dir/src/particles/deposition/PlasmaDepositCurrent.cpp.o] Error 1
```
I suspect this is due to #819.

--- a/src/particles/deposition/PlasmaDepositCurrent.cpp
+++ b/src/particles/deposition/PlasmaDepositCurrent.cpp
@@ -214,6 +214,9 @@ DepositCurrent (PlasmaParticleContainer& plasma, Fields & fields, const Laser& l
                     q_mu0_mass_ratio *= ion_lev[ip];
                 }
 
+                const amrex::Real xmid = (xp - x_pos_offset) * dx_inv;
+                const amrex::Real ymid = (yp - y_pos_offset) * dy_inv;
+
                 amrex::Real Aabssqp = 0._rt;
                 [[maybe_unused]] auto laser_arr = a_laser_arr;
                 if constexpr (use_laser.value) {
@@ -248,12 +251,10 @@ DepositCurrent (PlasmaParticleContainer& plasma, Fields & fields, const Laser& l
                         }
                         // --- Compute shape factors
                         // x direction
-                        const amrex::Real xmid = (xp - x_pos_offset) * dx_inv;
                         auto [shape_x, cell_x] =
                             compute_single_shape_factor<outer_depos_loop_x, depos_order_xy>(xmid, tx);
 
                         // y direction
-                        const amrex::Real ymid = (yp - y_pos_offset) * dy_inv;
                         auto [shape_y, cell_y] =
                             compute_single_shape_factor<false, depos_order_xy>(ymid, iy);
 

